### ISSUE TITLE
Supports OpenAPi spec v2.0 / v1.2

### ISSUE THREAD

**Issue by viclovsky** [OpenAPITools/openapi-diff#53]
Thanks a lot for a great tool. I really want to use it in my projects but it is impossible because there is no support of older versions of OpenAPi spec: 1.2 and 2.0. Is there any plans to fix it?

**Comment by quen2404** [comment]
Hello viclovsky,
this is not currently scheduled but you can create a merge request if you need it quickly. 
Else, i will try to do this by the end of week, but i'm not sure if i will have enough time to do that. And after, it's my holiday during two weeks.

**Comment by viclovsky** [comment]
Xd xd xc  oxkl

**Comment by quen2404** [comment]
@viclovsky i made a PR #53 to add the support but it seems there is a problem with `swagger-parser-v2-converter`. 
I opened an [issue](https://github.com/swagger-api/swagger-parser/issues/779) in this project.
But you can already try to use it if you checkout the branch `feature/support-swagger2`

**Comment by jmini** [comment]
This will be a big limitation of this approach: When parsing an old spec, you compare the result of the Swagger-Parser-Converter, which does an on the fly conversation to OAS3.

I work a lot with the Parser (and the converter module) for the [OpenAPI-Generator](https://github.com/OpenAPITools/openapi-generator) project and there are a lot corner cases that needs to be fixed... But it gets better and better (one issue at the time).

**Comment by quen2404** [comment]
Ok :) Thank you for your feedback @jmini !

**Comment by viclovsky** [comment]
Thanks a lot, 
Yep, I have already tried  it but sounds like I have some problems with it: it fails cause it can't translate my v2.0 spec into v3.0. Converter fails => diff fails.
1.  java.lang.IllegalArgumentException: Two path items have the same signature: `/some/api/method/{}`
But I don't have such signature at all!!! I have `/some/api/method/{id}, /some/api/method/{searchId},  /some/api/method/{id}/email `
2. java.lang.NullPointerException
 at io.swagger.v3.parser.converter.SwaggerConverter.convert(SwaggerConverter.java:422) 
If there is no 'info'  in specification

**Comment by quen2404** [comment]
You're contract seems invalid. 
1. `/some/api/method/{id}` & `/some/api/method/{id}` is the same rest endpoint.
2. According to [specification]((https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md), `info` is **required**

**Comment by quen2404** [comment]
Your contract isn't correctly designed.
You should have one path with two operations:
```yaml
paths:
  /some/api/method/{id}:
    GET:
      #...
    DELETE:
      #...
```

**Comment by jmini** [comment]
> 2. java.lang.NullPointerException
at io.swagger.v3.parser.converter.SwaggerConverter.convert(SwaggerConverter.java:422)
If there is no 'info' in specification

is already reported as https://github.com/swagger-api/swagger-parser/issues/755, will be fixed in the next version of Swagger-Parser.

**Comment by viclovsky** [comment]
> Your contract isn't correctly designed.
> You should have one path with two operations:
Yep, thanks a lot!! I've found https://github.com/swagger-api/swagger-editor/issues/1677 ... Thus I'm planning to change my spec to fit it ))

**Comment by jmini** [comment]
If you are interested, I have published an in-between release of `2.0.2-SNAPSHOT` that can be used until `2.0.2` is officially released:
https://github.com/OpenAPITools/swagger-parser/releases/tag/v2.0.2-OpenAPITools.org-1

**Comment by quen2404** [comment]
Hi @peterwa82, i can try again :) But last time, there was some issues in the project `swagger-parser-v2-converter`

**Comment by peterwa82** [comment]
There may still be issues with it.  I tried converting a 2.0 spec but noticed the required attribute of my query parameters was left null, which is wrong and results in a null pointer exception later.

**Comment by jmini** [comment]
@peterwa82 you need to report the issue you get in the https://github.com/OpenAPITools/swagger-parser project. With a simple example (OpenAPI v2.0) as input and with your expectation if the OpenAPI v3 is not the one you are expecting.

Here you will see a really simple java program to run swagger-parser independently:
https://j2r2b.github.io/2018/05/18/from-oas2-to-oas3.html

**Comment by joschi** [comment]
Fixed in 7734db057df34f5f5283fc00473fb50af3e1560d.
